features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/ElasticFlowRepository.java

```diff
@@ -155,6 +156,11 @@
      */
     private final Map<Direction, Cache<Integer, Set<Integer>>> markerCache = Maps.newEnumMap(Direction.class);
 
+    /**
+     * Collect flow documents ready for persistence.
+     */
+    private final Map<Thread, List<FlowDocument>> bulk = Maps.newHashMap();
```

Comment:
Should this be a concurrent hash map? We could have many threads adding new keys at the same time for example.

Reply:
done.